0.11 build fails on native arm(64|32) hosts

Hi,
I had 0.10 successfully building on arm64 but 0.11 is failing when npm installs or prunes dependencies.
I've tried on native 64-bit arm hardware as well as doing 32-bit builds under docker on arm64 (``docker run arm32v7/debian:buster``)

Also tried Debian Buster and Ubuntu 18.04 environments without docker.

Summary of issues:

- arm64 issue 1: optipng native build fails (tries to use neon (SIMD) acceleration which is not enabled in Debian's libpng): https://github.com/imagemin/optipng-bin/issues/97
If I modify dh_rules I can get around this:
```
-	$(NPM) --cache "$(NPM_CACHE)" install
+	CPPFLAGS="-DPNG_ARM_NEON_OPT=0" $(NPM) --cache "$(NPM_CACHE)" install

```
- arm64 issue 2: npm prune fails because gifsicle directory is not empty:
```
NODE_PATH=/usr/local/lib/node_modules/npm/node_modules /usr/local/lib/node_modules/npm/bin/npm-cli.js --cache "/tmp/tmp.BikbhqZfpP" prune --production
npm ERR! code ENOTEMPTY
npm ERR! syscall rmdir
npm ERR! path node_modules/gifsicle
npm ERR! errno -39
npm ERR! ENOTEMPTY: directory not empty, rmdir 'node_modules/gifsicle'
```
The gifsicle dir contains some strange file:
```~/gateway-deb/webthings-gateway/node_modules/gifsicle$ ls -la .
total 44
-rw-r--r--   1 ubuntu ubuntu     0 Feb 18 00:46 ''$'\001\f''3@@@'$'\321'
drwxr-xr-x   2 ubuntu ubuntu  4096 Feb 18 00:50  .
drwxr-xr-x 346 ubuntu ubuntu 36864 Feb 18 00:50  ..
```

I don't normally work with npm and node so I'm not sure how to go about debugging this...

- arm32 issue 1: prune fails (this time on mozjpeg):
```

npm ERR! code ENOTEMPTY
npm ERR! syscall rmdir
npm ERR! path node_modules/mozjpeg
npm ERR! errno -39
npm ERR! ENOTEMPTY: directory not empty, rmdir 'node_modules/mozjpeg'
make: *** [debian/rules:6: build] Error 2
```

Build logs (./build.sh and the npm logs) for both architectures in this gist:
https://gist.github.com/mcbridematt/cdccafb6f27fdcaffe318dbebfff8512

It looks like the CI builds are done using qemu-binfmt (binary emulation) on x86 hosts, my hunch is that 'native wrappers' like optipng could get through on the npm install step as they would be executed as native x86 binaries (especially when statically compiled). It looks like those get pruned anyway - none of the modules above are present in the Raspbian deb. (might be easier to blacklist them completely - is that possible?)

Modules that appear to be wrappers around binaries (do 'grep ELF: not found' on the build log)
- gifsicle
- jpegtrag-bin
- mozjpeg
- optipng-bin
- optipng
- pngquant
(All of the above try and recompile themselves if the prebuilt x86 binary fails to execute)

--- a/.github/workflows/pr.yml
+++ b/.github/workflows/pr.yml
@@ -17,6 +17,7 @@ jobs:
             "ubuntu:eoan",
             "debian:buster",
             "balenalib/raspberry-pi-debian:buster",
+            "balenalib/raspberrypi3-64-debian:buster",
           ]
         include:
           - image: "ubuntu:bionic"
@@ -34,6 +35,9 @@ jobs:
           - image: "balenalib/raspberry-pi-debian:buster"
             name: "raspbian-buster"
             arch: "arm"
+          - image: "balenalib/raspberrypi3-64-debian:buster"
+            name: "debian-buster"
+            arch: "arm64"
     steps:
       - name: Install qemu
         run: |

--- a/.github/workflows/push.yml
+++ b/.github/workflows/push.yml
@@ -17,6 +17,7 @@ jobs:
             "ubuntu:eoan",
             "debian:buster",
             "balenalib/raspberry-pi-debian:buster",
+            "balenalib/raspberrypi3-64-debian:buster",
           ]
         include:
           - image: "ubuntu:bionic"
@@ -34,6 +35,9 @@ jobs:
           - image: "balenalib/raspberry-pi-debian:buster"
             name: "raspbian-buster"
             arch: "arm"
+          - image: "balenalib/raspberrypi3-64-debian:buster"
+            name: "debian-buster"
+            arch: "arm64"
     steps:
       - name: Install qemu
         run: |

--- a/debian/rules
+++ b/debian/rules
@@ -11,8 +11,10 @@ override_dh_auto_clean:
 override_dh_auto_build:
 	$(eval NPM_CACHE := $(shell mktemp -dq))
 	$(RM) -rf ./node_modules package-lock.json
-	$(NPM) --cache "$(NPM_CACHE)" install
+	CPPFLAGS="-DPNG_ARM_NEON_OPT=0" $(NPM) --cache "$(NPM_CACHE)" install
 	./node_modules/.bin/webpack
+	$(RM) -rf ./node_modules/gifsicle
+	$(RM) -rf ./node_modules/mozjpeg
 	$(NPM) --cache "$(NPM_CACHE)" prune --production
 	$(RM) -rf "$(NPM_CACHE)"
 	chmod a+x webthings-gateway

